rust/exports.c

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-2.0
+//
+// A hack to export Rust symbols for loadable modules without having to redo
+// the entire `include/linux/export.h` logic.
+//
+// Note that this requires `-Z symbol-mangling-version=v0` because Rust's
+// default ("legacy") mangling scheme 1) uses a hash suffix which cannot
+// be predicted across compiler versions and 2) uses invalid characters
+// for C identifiers (thus we cannot use the `EXPORT_SYMBOL_*` macros).
+//
+// For the moment, we update this list manually. In the future, we should
+// have a way to handle this automatically.
+
+#include <linux/module.h>
+
+#define EXPORT_SYMBOL_RUST_GPL(sym) \
+	extern int sym; \
+	EXPORT_SYMBOL_GPL(sym);
+
+EXPORT_SYMBOL_RUST_GPL(__rust_dealloc);
+EXPORT_SYMBOL_RUST_GPL(__rust_alloc);
+EXPORT_SYMBOL_RUST_GPL(_RNvXsd_NtCsbDqzXfLQacH_4core3fmteNtB5_7Display3fmt);
```

Comment:
I think a comment for each of these with what it demangles to would be helpful.

Reply:
That's a good suggestion! At least meanwhile we need to maintain this list...

Reply:
Done.